lesson2/src/linked_list/listProblems.js

```diff
@@ -46,6 +88,19 @@ class ListProblems {
    */
   static detectCycle(list) {
     // todo: super magic 2 speed moving pointers ! 🎉
+
+    let node = list.head;
+
+
+    while (node) {
+      // Okay I've had help from the internet but could find no other way
+      if (node.seen) {
+        return true;
+      }
+      node.seen = true;
```

Comment:
haha yeah was not my intention, but this works amazing and its O(n) but in most cases you wont be able to modify the nodes (adding a visited flag 😔) no need to change it.. but give it a thought ... you can have 2 pointers (same as the example above) but instead of them being k nodes apart.. they start on the same node... but the fast pointer moves 2 nodes at a time `.next.next` if its a circle and one runner moves twice as fast as the other eventually the fast one will catch up again with the slow one right?